feat: Specific view for Dataset settings

keithCuniah opened the pull request:
# Description
Dataset settings feature 

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Not merge before:
- [x] #2210 
- [x] #2368 
- [x] #2369 
- [x] #2370 
- [x] #2371 
- [x] #2372 

Closes #2003 
Closes #2210 
Closes #2368 
Closes #2369 
Closes #2370 
Closes #2371 
Closes #2372 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] TextClassification
- [x] TokenClassification
- [x] Text2Text

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

Commit:
feat(Client): remove on mouseleave event

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

leiyre committed:
feat: dataset description component (#2414)

# Description

This PR includes new description component and some class to simplify
text styling

Closes #2372

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] DatasetDescription 

**Checklist**

- [x] I have merged the original branch into my forked branch
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'feature/front-dataset-settings' of https://github.com/recognai/rubrix into feature/front-dataset-settings

keithCuniah committed:
Feature : data label schema for labels management (#2384)

# Description
Fetch and store dataset labels as proper label schema

Closes #2210 

**Type of change**
- [x] remove obsolete code 
  -  _getDatasetSettings() in Dataset.js
  - initialize() in TextClassification.js and TokenClassification.js
  - initialize() call in the fetchByName() function in Dataset.js
- [x] get labels from api for TextClassification and TokenClassification
- [x] store labels as GlobalEntities in specific vuexorm (see "Ideal
Orm" in this
[link](https://www.notion.so/argilla/Refactoring-60c997e026644a28a2ff6a20c4cd90c3))
- [x] homogenized "AvailableRecords" (from TextClassificationHeader.vue)
and "visibleEntities"(from TokenClassificationHeader) by the same
vuexOrm query
- [x] prepare vuexorm queries to be able to sort globalLabels by any
param (by text, id, ...) and also ascending/descending
- [x] add more feedbacks with toast component if label could be saved,
there was a problem on save or if user try to add a label already saved



Note : 
- With these updates, globalLabels from tokenclassification and
textclassification have the same shape. M**eans they both have a
"color_id" param which can be used to add color for labels in
textClassification**
- the connection between globalEntities and dataset table is with the
"dataset_id" attribute from globalEntities. I didn't implement a
oneToMany relationship to not performance problem as in Weak labelling
implementation. This could be an idea of refactoring for later

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [x] Refactor (change restructuring the codebase without changing
functionality)
- [x] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] TokenClassification task
- [x] TextClassification task
- N/A Text2Text task
 
**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>

keithCuniah committed:
feat: create edition label component (#2422)

# Description
Edition label component

Please include a summary of the changes and the related issue. Please
also include relevant motivation and context. List any dependencies that
are required for this change.

To merge after #2210 

Closes #2371 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] Token classification
- [x] Text classification
- N/A Text2Text 

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>

leiyre committed:
feat: create Dataset delete component (#2430)

# Description

This PR includes new base-card component and dataset-delete component to
allow delete dataset from settings and remove the old logic from de
dataset list

Closes #2370

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] Test for BaseCard component
- [x] test for DatasetDeleteComponent

**Checklist**

- [x] I have merged the original branch into my forked branch
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' into feature/front-dataset-settings

keithCuniah committed:
Feature/create layout for dataset settings (#2448)

# Description
add generic layout for dataset settings page

Closes #2368 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] all task

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- N/A I have added tests that prove my fix is effective or that my
feature works

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>

leiyre committed:
feat: Go to dataset settings from dataset list (#2449)

# Description

This PR includes an icon in the dataset list to go to dataset settings

See [#2003](https://github.com/argilla-io/argilla/issues/2003)

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Checklist**

- [x] I have merged the original branch into my forked branch
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] My changes generate no new warnings

frascuchon committed:
chore: Merge branch 'develop' into feature/front-dataset-settings

frascuchon commented on frontend/database/modules/datasets.js:
```diff
@@ -21,6 +21,17 @@ import { AnnotationProgress } from "@/models/AnnotationProgress";
+import {
+  upsertLabelsInGlobalLabelModel,
+  deleteAllGlobalLabelModel,
+} from "../../models/globalLabel.queries";
```
...from "@/models/globalLabel.queries";?

frascuchon committed:
Add validation for manage settings by admins (#2450)

Adding extra custom validation allow only admin/superuser dataset
settings management.

If for example, a non-admin user tries to upgrade/update label schema,
this validation will return a 403 response.

frascuchon committed:
Merge branch 'develop' into feature/front-dataset-settings

frascuchon committed:
Change http method verb for settings (#2493)

Fix the settings VERB to PATCH since settings can be partially updated
(and deprecate the old one)

Commit:
Merge branch 'develop' of https://github.com/recognai/rubrix into develop

Commit:
Merge branch 'develop' into feature/front-dataset-settings

Commit:
feat: merge conflict

keithCuniah committed:
Feat : create page dataset settings (#2444)

# Description
create page dataset settings 

Please include a summary of the changes and the related issue. Please
also include relevant motivation and context. List any dependencies that
are required for this change.

Closes #2369 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] TokenClassification
- [x] TextClassification
- [x] Text2Text

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>
Co-authored-by: leire <leire@recogn.ai>

leiyre committed:
feat: QA from dataset settings (#2514)

# Description

This PR includes changes of styles and in the components logic after QA
IMPORTANT: Before merge this branch,
`feature/create-page-dataset-settings` must be merged

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>
Co-authored-by: keithCuniah <keithcuniah@gmail.com>

leiyre committed:
update test

frascuchon committed:
Fix missing method reference

Commit:
feat: get labels from GlobalLabelsORM

Commit:
Merge branch 'feature/front-dataset-settings' of https://github.com/recognai/rubrix into feature/front-dataset-settings

Commit:
feat: get all labels for weak labelling

Commit:
feat: update title button

leiyre committed:
refactor BaseFeedbackComponent in Edition Label Component

Commit:
feat: correction badge sync && destroy bus event

Commit:
Merge branch 'feature/front-dataset-settings' of https://github.com/recognai/rubrix into feature/front-dataset-settings

keithCuniah committed:
Update CHANGELOG.md

update changelog regarding dataset settings

keithCuniah committed:
Update CHANGELOG

frascuchon committed:
chore: Merge branch 'develop' into feature/front-dataset-settings
